### ISSUE TITLE
DO always removes some out-of-band changes (routes and managment-ip-rules)

### ISSUE THREAD

**Issue by megamattzilla** [F5Networks/f5-declarative-onboarding#328]
### Environment
 * Declarative Onboarding Version: 1.31
 * BIG-IP Version: 15.1.5.1

### Summary
DO always removes some out-of-band changes (routes and managment-ip-rules)

After DO has successfully processed a declaration and updated TMOS config, if you add some out of band configuration such as routes, management-ip-rules, user accounts some of these items will be deleted when DO runs again. 

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "schemaVersion": "1.31.0",
    "class": "Device",
    "async": true,
    "label": "my BIG-IP declaration for declarative onboarding",
    "Common": {
        "class": "Tenant",
        "mySystem": {
            "class": "System",
            "hostname": "bigip01.local",
            "autoCheck": true,
            "guiAuditLog": true,
            "mcpAuditLog": "enable",
            "tmshAuditLog": true,
            "autoPhonehome": false
        },
        "myProvisioning": {
                        "ltm": "nominal",
                        "asm": "nominal",
                        "afm": "nominal",
                        "class": "Provision"
        },
        "external": {
            "class": "VLAN",
            "mtu": 1500,
            "interfaces": [
                {
                "name": "1.2",
                "tagged": false
                }
            ],
            "cmpHash": "src-ip"
        },
        "internal": {
            "class": "VLAN",
            "mtu": 1500,
            "interfaces": [
                {
                "name": "1.1",
                "tagged": false
                }
            ],
            "cmpHash": "src-ip"
        },
        "external-self": {
            "class": "SelfIp",
            "address": "192.168.1.100/24",
            "vlan": "external",
            "trafficGroup": "traffic-group-local-only"
        },
        "internal-self": {
            "class": "SelfIp",
            "address": "192.168.2.100/24",
            "vlan": "internal",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "dbvars": {
            "class": "DbVariables",
            "setup.run": false,
            "kernel.pti": "disable",
            "provision.1nicautoconfig": "disable",
            "provision.extramb": 2048,
            "provision.tomcat.extramb": 20,
            "restjavad.useextramb": true,
            "ui.advisory.enabled": true,
            "ui.advisory.color": "red",
            "ui.system.preferences.recordsperscreen": "250",
            "ui.advisory.text": "bigip01 - ext_ip:192.168.1.100 int_ip:192.168.2.100 - Managed by Automation!"
        }
    }
}
```
2. Add out of band changes to users, route, and firewall management-ip-rules: 
```
admin@(bigip01)(cfg-sync Standalone)(ModuleNotLicensed::Active)(/Common)(tmos)# list auth user oob-user
auth user oob-user {
    description oob-user
    encrypted-password redacted
    partition Common
    partition-access {
        all-partitions {
            role guest
        }
    }
    session-limit -1
    shell none
}
admin@(bigip01)(cfg-sync Standalone)(ModuleNotLicensed::Active)(/Common)(tmos)# list net route
net route out-of-band-route {
    gw 192.168.1.1
    network 192.168.100.100/32
}
admin@(bigip01)(cfg-sync Standalone)(ModuleNotLicensed::Active)(/Common)(tmos)# list security firewall management-ip-rules
security firewall management-ip-rules {
    rules {
        oob-rule {
            action accept
            description "out of band change"
            ip-protocol any
            rule-number 1
        }
    }
}
```

3. Resubmit the same DO as step 1 (no changes) 

4. After DO successfully completes applying the DO declaration (no change) observe some of the TMOS changes are deleted: 
```
admin@(bigip01)(cfg-sync Standalone)(ModuleNotLicensed::Active)(/Common)(tmos)# list security firewall management-ip-rules
security firewall management-ip-rules { }
admin@(bigip01)(cfg-sync Standalone)(ModuleNotLicensed::Active)(/Common)(tmos)# list auth user oob-user
auth user oob-user {
    description oob-user
    encrypted-password redacted
    partition Common
    partition-access {
        all-partitions {
            role guest
        }
    }
    session-limit -1
    shell none
}
admin@(bigip01)(cfg-sync Standalone)(ModuleNotLicensed::Active)(/Common)(tmos)#(tmos)# list net route
<no output>
admin@(bigip01)(cfg-sync Standalone)(ModuleNotLicensed::Active)(/Common)(tmos)#
```
### Expected Behavior
After DO configuration is applied, out of band configuration such as routes (no routes specified in DO), firewall management-ip-rules (no rules specified in DO), and user accounts (none specified in DO) should remain present in TMOS. 

### Actual Behavior
After DO configuration is applied, some out of band configuration is removed such as routes (no routes specified in DO) and firewall management-ip-rules (no rules specified in DO). The OOB user accounts (none specified in DO) are still present in TMOS. 

Why are some out of band changes removed, while others OOB changes are still present? 
Is there a way to contain the blast radius of DO?

**Comment by sunitharonan** [comment]
Why are some out of band changes removed, while others OOB changes are still present?
Any out of band changes to state managed by Declarative onboarding will be removed.
https://clouddocs.f5.com/products/extensions/f5-declarative-onboarding/latest/schema-reference.html

Is there a way to contain the blast radius of DO? 
No

**Comment by seattlevine** [comment]
There is a bit more information on this DO behavior in the [docs](https://clouddocs.f5.com/products/extensions/f5-declarative-onboarding/latest/prereqs.html#notes-and-tipsl).

> The first time you POST a BIG-IP Declarative Onboarding declaration, the system records the configuration that exists prior to processing the declaration. BIG-IP Declarative Onboarding is meant to initially configure a BIG-IP device. However, if you POST subsequent declarations to the same BIG-IP system, and leave out some of the properties you initially used, the system restores the original properties for those items. Important: No matter what you send in a subsequent declaration, BIG-IP Declarative Onboarding will never unlicense a BIG-IP device, it will never delete a user, and it never break the device trust once it has been established. Thus, while BIG-IP Declarative Onboarding is declarative, it is not idempotent.

**Comment by megamattzilla** [comment]
We totally understand there is expected behavior where DO will remove things from TMOS when those things are removed from the DO declaration. 

That is not what's happening here. We are seeing DO remove config for TMOS configuration, which was never configured via DO. 

I deployed a fresh VM just to be sure. I submitted the DO declaration mentioned in step 1 and then configured an out of band static route and firewall management-ip-rules. These items are not defined in DO and never have been for this fresh VM. 

I then re-submitted the same DO declaration as step 1 (no changes) which was successfully applied. 

The out of band route and management firewall rule was removed by DO. I have never configured DO with the class route schema. 

Is that expected behavior?
